### PR TITLE
Add squid for customers.

### PR DESCRIPTION
**Pull request by kfyharukz**
Signed-off-by: Kazuhito MATSUDA <kazuhito.matuda@gmail.com>

### CODE REVIEW

**Review by morimoto-cybozu — APPROVED** [review]
LGTM

File: customer-egress/base/namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: customer-egress
+  annotations:
+    coil.cybozu.com/pool: customer-egress
+  labels:
+    control-plane: "true"
```
Comment (ymmt2005) [inline customer-egress/base/namespace.yaml:8]: add `team: neco`
https://github.com/cybozu-go/neco-apps/blob/master/namespaces/base/kustomization.yaml#L19-L20

File: customer-egress/base/neco/squid.yaml [hunk 2]
```diff
@@ -0,0 +1,360 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: squid
+  namespace: internet-egress
```
Comment (ymmt2005) [inline customer-egress/base/neco/squid.yaml:5]: customer-egress?

File: customer-egress/base/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: customer-egress
```
Comment (ymmt2005) [inline customer-egress/base/kustomization.yaml:3]: I don't think this is a good idea.
IMO, `namespace` stanza should only be used to fill in resources with missing namespace field.

**Review by ymmt2005 — CHANGES_REQUESTED** [review]

**Comment by ymmt2005** [comment]
@kfyharukz Please resolve the conflict too.

File: customer-egress/base/kustomization.yaml (hunk 3 above)
Comment (kfyharukz) [reply customer-egress/base/kustomization.yaml]: I see.  I've changed the maintenance way.
https://github.com/cybozu-go/neco-apps/pull/829/files#diff-0541ee302f10469f3b0290ea7585f45ff9652bc6de7131548819836586da0eb3R308

**Review by ymmt2005 — DISMISSED** [review]
LGTM

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Pull request merged**